win32/test/test_win32gui.py

```diff
@@ -61,5 +64,123 @@ def test_memory_not_writable(self):
+            self.assertIsNone(win32gui.EnumWindows(self.enum_callback, data))
+        for data in self.default_data_set:
+            self.assertRaises(
+                ValueError, win32gui.EnumWindows, self.enum_callback_exc, data
+            )
+        if sys.version_info[:2] >= (3, 10):
+            for func in (
+                self.enum_callback,
+                self.enum_callback_sle,
+            ):
+                for data in self.type_data_set:
+                    self.assertRaises(TypeError, win32gui.EnumWindows, func, data)
+
+    def test_enumchildwindows(self):
+        win32api.SetLastError(0)
+        for data in self.default_data_set:
+            self.assertIsNone(win32gui.EnumChildWindows(None, self.enum_callback, data))
+        for data in self.default_data_set:
+            self.assertIsNone(
+                win32gui.EnumChildWindows(None, self.enum_callback_sle, data)
+            )
+        win32api.SetLastError(0)
+        for data in self.default_data_set:
+            self.assertRaises(
+                ValueError,
+                win32gui.EnumChildWindows,
+                None,
+                self.enum_callback_exc,
+                data,
+            )
+        if sys.version_info[:2] >= (3, 10):
+            for data in self.type_data_set:
+                for func in (
+                    self.enum_callback,
+                    self.enum_callback_sle,
+                ):
+                    self.assertRaises(
+                        TypeError, win32gui.EnumChildWindows, None, func, data
+                    )
+
+    def test_enumdesktopwindows(self):
+        win32api.SetLastError(0)
+        desktop = None
+        for data in (0, False):
+            self.assertRaises(
+                pywintypes.error,
+                win32gui.EnumDesktopWindows,
+                desktop,
+                self.enum_callback_sle,
+                data,
+            )
+        for data in (None, 1, True):
+            self.assertIsNone(
+                win32gui.EnumDesktopWindows(desktop, self.enum_callback_sle, data)
+            )
+        win32api.SetLastError(0)
+        for data in self.default_data_set:
+            self.assertIsNone(
+                win32gui.EnumDesktopWindows(desktop, self.enum_callback, data)
+            )
+        for data in self.default_data_set:
+            self.assertRaises(
+                ValueError,
+                win32gui.EnumDesktopWindows,
+                desktop,
+                self.enum_callback_exc,
+                data,
+            )
+        desktops = (0, None)
+        for desktop in desktops:
+            for data in self.default_data_set:
+                self.assertRaises(
+                    ValueError,
+                    win32gui.EnumDesktopWindows,
+                    desktop,
+                    self.enum_callback_exc,
+                    data,
+                )
+        if sys.version_info[:2] >= (3, 10):
```

Comment:
Slicing is redundant here
```suggestion
        if sys.version_info[:2] >= (3, 10):
```

Reply:
done